### PR TITLE
NETOBSERV-1779 allow port filtering if protocol not set

### PR DESCRIPTION
**Pull request by jpinsonneau**
## Description

Allow port filtering if protocol not set

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [x] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Review by jotak — APPROVED** [review]

File: controllers/ebpf/agent_controller.go [hunk 1]
```diff
@@ -493,12 +461,40 @@ func (c *AgentController) configureFlowFilter(filter *flowslatest.EBPFFlowFilter
...
+	if filter.Ports.Type == intstr.String {
+		config = append(config, corev1.EnvVar{Name: envFilterPortRange,
+			Value: filter.Ports.String(),
+		})
+	}
+	if filter.Ports.Type == intstr.Int {
+		config = append(config, corev1.EnvVar{Name: envFilterPort,
+			Value: strconv.Itoa(filter.Ports.IntValue()),
+		})
+	}
```
Comment (msherif1234) [inline controllers/ebpf/agent_controller.go:493]: this won't work from ebpf agent processing the code look for the protocol then check the ports so if protocol not set it won't try to match the ports

Comment (jpinsonneau) [reply controllers/ebpf/agent_controller.go:493]: Can't we catch all protocols in that case ? :thinking: 
I thought that was the output of our conversations on that one

If not, I'll fall back on your suggestion about an error in FlowCollector status

Comment (msherif1234) [reply controllers/ebpf/agent_controller.go:493]: we can't make a guess about the protocol when we parse the pkt in ebpf so not sure how will the catch all works lets chat about it so we both are clear ?

Comment (msherif1234) [reply controllers/ebpf/agent_controller.go:493]: looks now I updated the agent to work in this mode pls bring this fix to 1.6.1

File: controllers/ebpf/agent_controller.go [hunk 2]
```diff
@@ -449,38 +449,6 @@ func (c *AgentController) configureFlowFilter(filter *flowslatest.EBPFFlowFilter
...
-					Value: filter.Ports.String(),
-				})
-			}
-			if filter.Ports.Type == intstr.Int {
-				config = append(config, corev1.EnvVar{Name: envFilterPort,
-					Value: strconv.Itoa(filter.Ports.IntValue()),
-				})
-			}
-
 		case "ICMP", "ICMPv6":
```
Comment (msherif1234) [inline controllers/ebpf/agent_controller.go:452]: lets move icmp filters out of this switch statement and remove the switch statement

Comment (jpinsonneau) [reply controllers/ebpf/agent_controller.go]: sure https://github.com/netobserv/network-observability-operator/pull/738/commits/4422d58e4f1ab046018a267ec94666b2897d2715

**Comment by Amoghrd** [comment]
The filtering seems to be broken in this PR. Upon adding Reject rule without Protocol for SrcPort 53, I still see flows from that port.

**Comment by msherif1234** [comment]
yes this should be fixed by https://github.com/netobserv/network-observability-operator/pull/754 I introduced this issue

**Comment by Amoghrd** [comment]
@jpinsonneau Could you rebase this PR?

**Pull request merged**